clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/model/ElasticSearchEntityTagsProvider.java

```diff
@@ -496,8 +502,38 @@ public Map reconcile(String cloudProvider, String account, String region, boolea
       results.put("deletedFromElasticsearch", true);
 
       if (deleteFromSource) {
-        Lists.partition(entityTagsForNamespace, 50).forEach(front50Service::batchUpdate);
-        results.put("deletedFromSource", true);
+        ExecutorService executor =
+            Executors.newFixedThreadPool(
+                6,
+                new ThreadFactoryBuilder()
+                    .setNameFormat(ElasticSearchEntityTagsProvider.class.getSimpleName() + "-%d")
+                    .build());
+
+        AtomicLong countProcessed = new AtomicLong();
+
+        Lists.partition(entityTagsForNamespace, 50)
+            .forEach(
+                entityTags ->
+                    executor.submit(
+                        () -> {
+                          AuthenticatedRequest.allowAnonymous(
+                              () -> front50Service.batchUpdate(entityTags));
+
+                          log.info(
+                              "Deleted {} out of {} tags in namespace {}",
+                              countProcessed.addAndGet(entityTags.size()),
+                              entityTagsForNamespace.size(),
+                              namespace);
+                        }));
+
+        try {
+          executor.shutdown();
+          executor.awaitTermination(1, TimeUnit.HOURS);
```

Comment:
this seems like a long upper bounds time to block on this request, is this close to on the order of how long cleanup takes?

Reply:
it's how long it used to take, now i think it should be much smaller (in my attempts/estimates <10 mins, i will reduce it to 15 mins and a note to the exception for the operator to retry )